packages/pglite/src/fs/base.ts

```diff
@@ -0,0 +1,547 @@
+import type { PostgresMod } from '../postgresMod.js'
+import type { PGlite } from '../pglite.js'
+import { dumpTar, type DumpTarCompressionOptions } from './tarUtils.js'
+
+export const WASM_PREFIX = '/tmp/pglite'
+export const PGDATA = WASM_PREFIX + '/' + 'base'
+
+export type FsType =
+  | 'nodefs'
+  | 'idbfs'
+  | 'memoryfs'
+  | 'opfs-worker'
+  | 'opfs-ahp'
+
+/**
+ * Filesystem interface.
+ * All virtual filesystems that are compatible with PGlite must implement
+ * this interface.
+ */
+export interface Filesystem {
+  /**
+   * Initiate the filesystem and return the options to pass to the emscripten module.
+   */
+  init(
+    pg: PGlite,
```

Comment:
Yep, future proofing. If a VFS wants to access some other methods on the PGlite instance in future it can. I did have a use case in mind but I forget it right now.